adambrg/src/main/java/ServletAdd.java

```diff
@@ -0,0 +1,39 @@
+import jakarta.servlet.ServletException;
+import jakarta.servlet.annotation.WebServlet;
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet("/user/add")
+public class ServletAdd extends HttpServlet {
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        getServletContext().getRequestDispatcher("/users/userAdd.jsp").forward(request, response);
+
+    }
+
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+        String userName = request.getParameter("userName");
```

Comment:
CLEAN-CODE: pouzivat nemenne promenne vsude kde to je mozne v aplikaci.

```suggestion
final String username = req.getParameter("username");
```

**Duvod**

Používání neměnných proměnných final v Javě je dobré, protože zajišťuje, že hodnota proměnné se po jejím přiřazení nemůže změnit, což zvyšuje čitelnost a spolehlivost kódu. Pomáhá to také při optimalizaci kódu, protože kompilátor může lépe předvídat chování programu. Dále zajišťuje, že nebudou neúmyslné změny hodnoty proměnných, což snižuje riziko chyb, a také to usnadňuje práci s multithreadingem, kde je důležité, aby hodnoty nebyly modifikovány souběžně v různých vláknech.

Reply:
opraveno